lib/vocabularies/validation/pattern.ts

```diff
@@ -18,9 +18,16 @@ const def: CodeKeywordDefinition = {
   error,
   code(cxt: KeywordCxt) {
     const {data, $data, schema, schemaCode, it} = cxt
-    // TODO regexp should be wrapped in try/catchs
     const u = it.opts.unicodeRegExp ? "u" : ""
-    const regExp = $data ? _`(new RegExp(${schemaCode}, ${u}))` : usePattern(cxt, schema)
+    const regExp = $data
+      ? _`(function() {
```

Comment:
There is probably a way to build this with `gen.try` etc but maybe this is ok? I'm still getting my head around codegen stuff.